src/modules/home/components/accommodation-card.tsx

```diff
@@ -57,12 +46,61 @@ const styles = {
   ratingStar: { fontSize: '17px' },
 };
 
-export function AccommodationCard({ status, id, pricePerNight, address, name, rating, image }: AccommodationCardProps) {
-  const [addToWishlist, setAddToWishlist] = useState(true);
+export function AccommodationCard({
+  status,
+  id,
+  pricePerNight,
+  address,
+  name,
+  rating,
+  image,
+  isSavedToWishlist,
+}: AccommodationCardProps) {
+  const [isWishlisted, setIsWishlisted] = useState(isSavedToWishlist);
```

Comment:
I don't see why we need this state. The data should be returned from the backend

Reply:
> I don't see why we need this state. The data should be returned from the backend

It is returned from backend and the state is initialized with the value returned from backend. How can I handle the heart button color change and which action to send when clicked without state?